File: seminar/src/main/java/com/server/dosopt/seminar/dto/request/post/ValidTitle.java
package com.server.dosopt.seminar.dto.request.post;

import jakarta.validation.Constraint;
import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Documented
@Constraint(validatedBy = TitleValidator.class)
@Target({java.lang.annotation.ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
public @interface ValidTitle {

    String message() default "유효하지 않은 제목입니다.";

    String pattern() default "[가-힣|a-z|A-Z|0-9|]";
}

File: kotlin-seminar/src/main/kotlin/sopt/org/kotlinseminar/dto/request/post/ValidTitle.kt
package sopt.org.kotlinseminar.dto.request.post

import jakarta.validation.Constraint


@MustBeDocumented
@Constraint(validatedBy = [TitleValidator::class])
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class ValidTitle(
        val message: String = "Invalid title",
        val pattern: String = "[가-힣|a-z|A-Z|0-9|]"
)